rules/TypeDeclaration/NodeAnalyzer/ClassMethodParamTypeCompleter.php

```diff
@@ -49,6 +52,18 @@ public function complete(ClassMethod $classMethod, array $classParameterTypes, i
                 continue;
             }
 
+            if ($phpParserTypeNode instanceof IntersectionType) {
+                $types = $phpParserTypeNode->types;
+
+                $firstType = (string) $this->nodeNameResolver->getName($types[0]);
+                $secondType = (string) $this->nodeNameResolver->getName($types[1]);
+
+                if ($firstType === $secondType) {
+                    // choose one
+                    $phpParserTypeNode = $types[0];
+                }
+            }
```

Comment:
This looks weird. The intersectiontype should be probably narrowed to single one.

Reply:
I moved check to IntersectionTypeMapper with check unique https://github.com/rectorphp/rector-src/pull/1375/commits/3cbb58ee08353f9a6753b59adff2a0fc665ab472